### PR TITLE
cmd_search: Add support for user@service queries

### PR DESCRIPTION
**Pull request by shazow**
This adds support for doing queries like `keybase search shazow@twitter` by adding a `service` parameter to the user search function in the avdl.

The server endpoint already supports the "service" param, we just weren't passing it through.

### CODE REVIEW

File: go/protocol/keybase1/user.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ type ListTrackingJSONArg struct {
 type SearchArg struct {
 	SessionID int    `codec:"sessionID" json:"sessionID"`
 	Query     string `codec:"query" json:"query"`
+	Service   string `codec:"service" json:"service"`
```
Comment (shazow) [inline go/protocol/keybase1/user.go]: Are there any problems/risks with changing the protocol subpackage?

Comment (oconnor663) [inline go/protocol/keybase1/user.go]: Yes, all of that is generated code. You'll want to change [the AVDL file it came from](https://github.com/keybase/client/blob/master/protocol/avdl/keybase1/user.avdl), and rerun `make` in `./protocol/`.

Comment (shazow) [inline go/protocol/keybase1/user.go]: Oh right I remember seeing the header, oops.

**Comment by oconnor663** [comment]
You're adding an extra field to `SearchArgs`, but don't you need to make sure that extra field gets used somehow? I'm looking at where the first field gets used, here: https://github.com/keybase/client/blob/master/go/service/user.go#L139

**Comment by shazow** [comment]
Looks like the queries get [serialized here](https://github.com/keybase/client/blob/master/go/protocol/keybase1/user.go#L546) rather than via the service runner thing.

This PR works locally fwiw.

File: go/protocol/keybase1/user.go (hunk 1 above)
Comment (shazow) [inline go/protocol/keybase1/user.go]: Fixed.

Comment (shazow) [inline go/protocol/keybase1/user.go]: Ok now properly re-ran it. Will find another person to review once I confirm everything is still working. :)

**Comment by maxtaco** [comment]
cc: @oconnor663 ping

**Comment by shazow** [comment]
@maxtaco Jack already took a look but would prefer another person to check it out. Maybe @mmaxim?

**Comment by patrickxb** [comment]
This doesn't actually change the service.

**Comment by patrickxb** [comment]
```func (h *UserHandler) Search(_ context.Context, arg keybase1.SearchArg) (results []keybase1.SearchResult, err error) {
	eng := engine.NewSearchEngine(engine.SearchEngineArgs{
		Query: arg.Query,
	}, h.G())
	ctx := &engine.Context{
		LogUI:     h.getLogUI(arg.SessionID),
		SessionID: arg.SessionID,
	}
	err = engine.RunEngine(eng, ctx)
	if err == nil {
		results = eng.GetResults()
	}
	return
}
```

**Comment by oconnor663** [comment]
LGTM as far as the code goes, though I'm still woefully confused about how it actually works. (It looks like the `SearchArg` struct makes it to `go/service/user.go::UserHandler::Search`, and there the `Service` param gets ignored. I think Patrick's asking the same thing.)

**Comment by patrickxb** [comment]
service/user.go:Search.

You'll need to change that to plumb through the service parameter.

**Comment by patrickxb** [comment]
engine/search.go needs to change too.

**Comment by shazow** [comment]
Chatted with Patrick, looks like it's using the `user/autocomplete` endpoint but we could switch it to use `user/user_search` which might be better—would need to confirm that the frontend is not using this anywhere.

Also no clue how my local version worked at all given that it wasn't plumbed through.

I'm going to put this PR on hold for a bit while I resume work on some extension frontend stuff.

**Pull request closed without merging**